### PR TITLE
Cache token server request for the availability of the hawk credentials duration

### PR DESCRIPTION
**Pull request by Natim**
- [x] Cache credentials

### CODE REVIEW

**Comment by michielbdejong** [comment]
What are the security implications? If an attacker gets access to a Syncto server during 10 minutes, it might impact only a few users if we don't cache, but thousands of users if we do?

**Comment by Natim** [comment]
If an attacker gets access to a Syncto server it is bad anyway but the only thing that you can do is to delete everything because everything is encrypted.

A Sync Client feature is to reupload everything if we detect that the data has been removed.

So the security impact is quite small, also @jvehent is going to have a closer look at security problems for Syncto.

File: syncto/tests/test_authentication.py [hunk 1]
```diff
@@ -23,6 +37,26 @@ def test_should_raise_if_client_state_header_is_missing(self):
...
+        with mock.patch('syncto.authentication.TokenserverClient') as TSClient:
+            TSClient.return_value.get_hawk_credentials.return_value = \
+                self.credentials
+            with mock.patch('syncto.authentication.SyncClient') as SyncClient:
+                # First call
+                build_sync_client(self.request)
+                SyncClient.assert_called_with(**self.credentials)
+                # Second time
+                build_sync_client(self.request)
+                TSClient.assert_called_once_with('1234', '12345')
```
Comment (leplatrem) [inline syncto/tests/test_authentication.py:63]: nit: At first look, I didn't see the `_once_`. Maybe emphasize with a comment ?

File: syncto/__init__.py [hunk 2]
```diff
@@ -17,10 +17,15 @@
 AUTHORIZATION_HEADER = 'Authorization'
 CLIENT_STATE_HEADER = 'X-Client-State'
 
+DEFAULT_SETTINGS = {
+    'syncto.cache_hmac_secret': 'This is not a secret',
```
Comment (almet) [inline syncto/__init__.py]: I'm not sure we want to have a default value for this. We should rather error out if it's not defined.

**Comment by almet** [comment]
LGTM with the removal of the default hmac secret.

File: syncto/__init__.py (hunk 2 above)
Comment (phrawzty) [inline syncto/__init__.py]: Agreed - the only sane default is to prevent unsafe usage, imho.

**Pull request merged**